core/tx_pool.go

```diff
@@ -535,7 +536,11 @@ func (pool *TxPool) validateTx(tx *types.Transaction, local bool) error {
 	if pool.currentMaxGas < tx.Gas() {
 		return ErrGasLimit
 	}
-	// Make sure the transaction is signed properly
+	// Accept only legacy transactions if before 2718/2930.
+	if !pool.eip2718 && tx.Type() != types.LegacyTxType {
+		return ErrTxTypeNotSupported
+	}
```

Comment:
Imho this check should be the first one. Don't even look any any of the fields for an unsupported tx type.